components/support/locale/src/main/java/mozilla/components/support/locale/LocaleAwareAppCompatActivity.kt

```diff
@@ -15,4 +17,15 @@ open class LocaleAwareAppCompatActivity : AppCompatActivity() {
         val context = LocaleManager.updateResources(base)
         super.attachBaseContext(context)
     }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // Android 8 has a bug which doesn't change the layoutDirection on activity recreation.
+        // https://github.com/mozilla-mobile/fenix/issues/9413
+        // https://stackoverflow.com/questions/46296202/rtl-layout-bug-in-android-oreo#comment98890942_46298101
+        if (Build.VERSION.SDK_INT == Build.VERSION_CODES.O) {
```

Comment:
It looks good, lets just add a test for it :)

Reply:
🤔 I started to go down that route. Do we have a place where we test build specific things already? I'm not sure how to do that on Android

Reply:
[Robolectric allows to customized](http://robolectric.org/configuring/) the version of Android using  `@Config(sdk =[version])` :) 

Throughout the source code, we use it in a couple of places take a look at [SecureAbove22PreferencesTest]( https://github.com/mozilla-mobile/android-components/blob/afb265c3b7aff459066b041b12cc56068df773c0/components/lib/dataprotect/src/test/java/mozilla/components/lib/dataprotect/SecureAbove22PreferencesTest.kt#L20) 

If you need any additional help just let me know :)